docs/guides/gitlab.md

```diff
@@ -0,0 +1,268 @@
+---
+title: GitLab
+lang: en-US
+meta:
+  - name: keywords
+    content: >-
+      pomerium identity-access-proxy gitlab gitlab-ee docker
+      authentication authorization
+description: >-
+  This guide covers how to secure self-hosted GitLab behind Pomerium, providing authentication and authorization through your IdP.
+---
+
+# GitLab
+
+[GitLab] is a highly customizable, highly configurable tool to manage source code, project management, and many other aspects of project development. In addition to the SaaS product, its self-hosted solution and easy free-to-enterprise upgrade path make it a popular choice for those managing sensitive code bases.
+
+This guide demonstrates how to configure a self-hosted GitLab server (a.k.a. gitlab-ee) behind Pomerium for identity-aware access.
+
+## Before You Being
+
+- This guide is written for an environment using [Docker Compose].
+
+- This guide assumes a running instance of Pomerium, already configured with an identity Provider (**IdP**) and running as a docker container on the same host/swarm. See our [Quick-Start] page for more information on installing Pomerium with Docker Compose.
+
+  ::: warning
+  While Pomerium can be configured to use self-hosted GitLab [as an IdP][gitlab-idp], we do not recommend doing so while also running it behind Pomerium. In addition to the potential to lock out access to the IdP (breaking access to all routes), we consider it best practice to keep a separation of services between your identity provider protected services, especially those housing sensitive data like source code.
```

Comment:
I'm not opposed to changing the phrasing, but this change suggests that the services are protected by the IdP, when they are not, which I see now that the phrase indicated as is. The original should have said:
```suggestion
  While Pomerium can be configured to use self-hosted GitLab [as an IdP][gitlab-idp], we do not recommend doing so while also running it behind Pomerium. In addition to the potential to lock out access to the IdP (breaking access to all routes), we consider it best practice to keep a separation of services between your identity provider and protected services, especially those housing sensitive data like source code.
```
Do you want to take a pass on this fixed version?

Reply:
```suggestion
  While Pomerium can be configured to use self-hosted GitLab [as an IdP][gitlab-idp], we do not recommend doing so while also running it behind Pomerium. In addition to the potential to lock out access to the IdP (breaking access to all routes), we consider it best practice to separate your identity provider and protected services, especially those housing sensitive data like source code.
```
I think this would work assuming I understood the intent.